src/rendering/framebufferrenderer.cpp

```diff
@@ -66,955 +66,963 @@ namespace {
-    GLint defaultFbo;
-    glGetIntegerv(GL_FRAMEBUFFER_BINDING, &defaultFbo);
+    void FramebufferRenderer::render(float blackoutFactor, bool doPerformanceMeasurements) {
+        if (_dirtyResolution) {
+            updateResolution();
+        }
 
-    glBindFramebuffer(GL_FRAMEBUFFER, _mainFramebuffer);
-    // deferred g-buffer
-    GLenum textureBuffers[3] = {
-        GL_COLOR_ATTACHMENT0,
-        GL_COLOR_ATTACHMENT1,
-        GL_COLOR_ATTACHMENT2,
-    };
-    glDrawBuffers(3, textureBuffers);
-    glClear(GL_COLOR_BUFFER_BIT | GL_DEPTH_BUFFER_BIT);
+        if (_dirtyMsaaSamplingPattern) {
+            updateMSAASamplingPattern();
+        }
 
-    glEnablei(GL_BLEND, 0);
-    glDisablei(GL_BLEND, 1);
-    glDisablei(GL_BLEND, 2);
-    glBlendFunc(GL_SRC_ALPHA, GL_ONE_MINUS_SRC_ALPHA);
+        if (_dirtyRaycastData) {
+            updateRaycastData();
+        }
 
-    {
-        std::unique_ptr<performance::PerformanceMeasurement> perfInternal;
+        if (_dirtyDeferredcastData) {
+            updateDeferredcastData();
+        }
+
+        std::unique_ptr<performance::PerformanceMeasurement> perf;
         if (doPerformanceMeasurements) {
-            perfInternal = std::make_unique<performance::PerformanceMeasurement>(
-                "FramebufferRenderer::render::background",
+            perf = std::make_unique<performance::PerformanceMeasurement>(
+                "FramebufferRenderer::render",
                 OsEng.renderEngine().performanceManager()
                 );
         }
 
-        data.renderBinMask = static_cast<int>(Renderable::RenderBin::Background);
-        _scene->render(data, tasks);
-    }
-    {
-        std::unique_ptr<performance::PerformanceMeasurement> perfInternal;
-        if (doPerformanceMeasurements) {
-            perfInternal = std::make_unique<performance::PerformanceMeasurement>(
-                "FramebufferRenderer::render::opaque",
-                OsEng.renderEngine().performanceManager()
-            );
+        if (!_scene || !_camera) {
+            return;
         }
 
+        // Capture standard fbo
+        GLint defaultFbo;
+        glGetIntegerv(GL_FRAMEBUFFER_BINDING, &defaultFbo);
+
+        glBindFramebuffer(GL_FRAMEBUFFER, _mainFramebuffer);
+        glEnable(GL_DEPTH_TEST);
+
+        // deferred g-buffer
+        GLenum textureBuffers[3] = {
+            GL_COLOR_ATTACHMENT0,
+            GL_COLOR_ATTACHMENT1,
+            GL_COLOR_ATTACHMENT2,
+        };
+        glDrawBuffers(3, textureBuffers);
+        glClear(GL_COLOR_BUFFER_BIT | GL_DEPTH_BUFFER_BIT);
+
+        glEnablei(GL_BLEND, 0);
+        glDisablei(GL_BLEND, 1);
+        glDisablei(GL_BLEND, 2);
+        glBlendFunc(GL_SRC_ALPHA, GL_ONE_MINUS_SRC_ALPHA);
+
+        Time time = OsEng.timeManager().time();
+
+        RenderData data = { *_camera, psc(), time, doPerformanceMeasurements, 0,{} };
+        RendererTasks tasks;
+
+        data.renderBinMask = static_cast<int>(Renderable::RenderBin::Background);
```

Comment:
Why isn't the `renderBinMask` of type `Renderable::RenderBin`?

Reply:
No, it is an int.

Reply:
I guess that's more a question for @emiax 

Why is it an int?  to support bitmasking?